src/CriticalSection/Driver/PdoMysqlDriver.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace stekycz\CriticalSection\Driver;
+
+use PDO;
+
+/**
+ * @see https://mariadb.com/kb/en/mariadb/get_lock
+ * @see https://dev.mysql.com/doc/refman/5.7/en/miscellaneous-functions.html#function_get-lock
+ */
+class PdoMysqlDriver implements IDriver
+{
+	const NO_WAIT = 0;
+
+	/**
+	 * @var PDO
+	 */
+	private $pdo;
+
+	/**
+	 * @var int
+	 */
+	private $lockTimeout;
+
+	public function __construct(PDO $pdo, int $lockTimeout = self::NO_WAIT)
+	{
+		$this->pdo = $pdo;
+		$this->lockTimeout = $lockTimeout;
+	}
+
+	public function acquireLock(string $label) : bool
+	{
+		return $this->runQuery('SELECT GET_LOCK(?, ?)', $label, $this->lockTimeout);
```

Comment:
MySQL version >= 5.7.5 specifies that string for lock (`$label`) must have up to 64 characters (I assume data type `char`). However previous version were unlimited and MariaDB does not say a thing about any limitation.

Is it save to keep it this way because MySQL triggers a PDO exception that would be propagated up in call stack? Or is better idea to make always a hash of `$label`, let's say `sha1`? I have to use `crc32` in `PdoPgsqlDriver` because PgSQL requires integer either so I would prefer using hashing to keep it working out of the box for everyone.